test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+ * 4. Check that the corresponding topic has been created and the type has been registered.
+ * 5. Enable and already enabled statistics DataWriter and check that it returns RETCODE_OK.
+ * 6. Call the method with an invalid topic name and check that returns RETCODE_BAD_PARAMETER.
+ * 7. Disable one statistics DataWriter and check that it is successful.
+ * 8. Enable the previous statistics DataWriter with an inconsistent QoS and check that it returns
+ * RETCODE_INCONSISTENT_POLICY.
+ * The case where the create_datawriter fails returning RETCODE_ERROR is not checked because it only passes the error
+ * upstream.
+ * 9. Try to delete the DomainParticipant and check that fails with RETCODE_PRECONDITION_NOT_MET because there are still
+ * statistics DataWriters enabled.
+ * 10. Try to disable an already disabled statistics DataWriter and check that returns RETCODE_ERROR.
+ * 11. Check that if an invalid topic name is provided to the disable_statistics_datawriter method, it returns
+ * RETCODE_BAD_PARAMETER.
+ * 12. Delete DDS entities.
+ * The case where the delete_datawriter fails returning RETCODE_ERROR is not checked because it only passes the error
+ * upstream.
+ */
+TEST(StatisticsDomainParticipantTests, EnableDisableStatisticsDataWriterTest)
+{
+eprosima::fastdds::dds::DataWriterQos inconsistent_qos = eprosima::fastdds::dds::DATAWRITER_QOS_DEFAULT;
+inconsistent_qos.reliability().kind = eprosima::fastdds::dds::BEST_EFFORT_RELIABILITY_QOS;
+inconsistent_qos.ownership().kind = eprosima::fastdds::dds::EXCLUSIVE_OWNERSHIP_QOS;
+
+#ifndef FASTDDS_STATISTICS
+    // 1. Compilation flag not set
+    eprosima::fastdds::statistics::dds::DomainParticipant statistics_participant;
+    eprosima::fastrtps::types::ReturnCode_t ret = statistics_participant.enable_statistics_datawriter(
+            HISTORY_LATENCY_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+
+    ret = statistics_participant.enable_statistics_datawriter(
+            "INVALID_TOPIC", inconsistent_qos);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+
+    ret = statistics_participant.disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+
+    ret = statistics_participant.disable_statistics_datawriter("INVALID_TOPIC");
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+#else
+    logError(STATISTICS_DOMAINPARTICIPANT_TEST, "This test is going to fail because API is not yet implemented.")
+
+    // 2. Create DomainParticipant and narrow to eprosima::fastdds::statistics::dds::DomainParticipant
+    eprosima::fastdds::dds::DomainParticipant * participant =
+            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            create_participant(0, eprosima::fastdds::dds::PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
```

Comment:
Move to the beginning of the test

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3